### PR TITLE
Merge Path and Operation parameters

### PR DESCRIPTION
**Pull request by saharj**
Fixes https://github.com/swagger-api/swagger-js/issues/965

### CODE REVIEW

**Comment by saharj** [comment]
@ponelat

File: src/execute.js [hunk 1]
```diff
@@ -86,8 +86,11 @@ export function buildRequest({
   if (requestContentType) {
     req.headers['content-type'] = requestContentType
   }
+
   // Add values to request
-  arrayOrEmpty(operation.parameters).forEach((parameter) => {
+  arrayOrEmpty(operation.parameters) // operation parameters
+  .concat(arrayOrEmpty(spec.paths[pathName].parameters)) // path parameters
```
Comment (buunguyen) [inline src/execute.js]: Please add indent to this and the next line

File: test/execute.js [hunk 2]
```diff
@@ -1196,6 +1196,80 @@ describe('execute', () => {
...
+                  in: 'path',
+                  type: 'number',
+                  required: true
+                }
+              ],
+            }
+          }
+        }
+
+        const req = buildRequest({ spec, operationId: 'getPetsById', parameters: { id: 123, test: 567 } })
```
Comment (buunguyen) [inline test/execute.js]: Please remove empty space inside `{}`.

File: src/helpers.js [hunk 3]
```diff
@@ -60,6 +60,9 @@ export function eachOperation(spec, cb, find) {
   // Iterate over the spec, collecting operations
   for (const pathName in paths) {
     for (const method in paths[pathName]) {
+      if (method.toUpperCase() === 'PARAMETERS') {
```
Comment (buunguyen) [inline src/helpers.js]: What is this check for? Would be great to add a comment clarifying the intention.

Comment (saharj) [inline src/helpers.js]: Oddly, when path-parameters come before the operation like below, the method becomes `PARAMETERS` (here it should be `GET`), and that causes problems.
```
paths: {
  '/pet/{id}': {
    parameters: [
      {...}
    ],
    get: {
      operationId: 'getPetsById',
      parameters: [
        {...}
      ],
    }
  }
}
```

Comment (buunguyen) [inline src/helpers.js]: I see. The loop iterates the keys of each path, so "parameters" will appear. Doesn't matter if it appears before or after the "get".

**Pull request merged**